envoy.code.check/envoy/code/check/typing.py

```diff
@@ -19,6 +19,7 @@ class BaseExtensionMetadataDict(TypedDict):
     categories: List[str]
     security_posture: str
     status: str
+    status_upstream: str
```

Comment:
I think we only want to require status_upstream if one of the categories has the string literal "http.upstream"

Reply:
cool ill check the check logic to see if we can accomodate - and add some commenting about per-cat status

Reply:
> I think we only want to require

i guess that goes for allow also

Reply:
indeed!